plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -43,6 +44,9 @@
 # Use a custom logging handler to display messages in Maya script editor.
 standalone_logger.addHandler(plugin_logging.PluginLoggingHandler(manifest.name))
 
+# Main progress bar used when bootstrapping the toolkit and engine.
+main_progress_bar = None
```

Comment:
After some experimentation, now using a variation of this approach, with no more global since there is only one main progress bar.

In between plug-in progress bar updates, other user interactions may overwrite the progress message, and possibly the progress bar. At the next plug-in update, the new code makes sure the progress bar uses the proper configuration and updates the progress value and message.